test/e2e_node/container_lifecycle_test.go

```diff
@@ -398,6 +398,215 @@ var _ = SIGDescribe(framework.WithNodeConformance(), "Containers Lifecycle", fun
+					containerStatus := pod.Status.ContainerStatuses[0]
+					return containerStatus.State.Terminated != nil && containerStatus.State.Terminated.ExitCode == 0, nil
+				})
+				framework.ExpectNoError(err)
+
+				podSpec, err = client.Get(ctx, podSpec.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+				results := parseOutput(ctx, f, podSpec)
+				framework.ExpectNoError(results.HasNotRestarted(regular1))
+			})
+		})
+
+		ginkgo.It("should not restart with restartPolicy=OnFailure", func(ctx context.Context) {
+			originalPodSpec.Spec.RestartPolicy = v1.RestartPolicyOnFailure
+			client := e2epod.NewPodClient(f)
+			podSpec := client.Create(ctx, originalPodSpec)
+
+			ginkgo.By("running the pod", func() {
+				err := e2epod.WaitForPodNameRunningInNamespace(ctx, f.ClientSet, podSpec.Name, podSpec.Namespace)
+				framework.ExpectNoError(err)
+			})
+
+			ginkgo.By("updating the image", func() {
+				client.Update(ctx, podSpec.Name, func(pod *v1.Pod) {
+					podSpec.Spec.Containers[0].Image = imageutils.GetE2EImage(imageutils.InvalidRegistryImage)
+
+				})
+
+			})
+
+			ginkgo.By("verifying that the container does not restart", func() {
+				err := e2epod.WaitForPodSuccessInNamespace(ctx, f.ClientSet, podSpec.Name, podSpec.Namespace)
+				framework.ExpectNoError(err)
+
+				err = e2epod.WaitForPodCondition(ctx, f.ClientSet, podSpec.Namespace, podSpec.Name, "the container terminated regularly", 30*time.Second, func(pod *v1.Pod) (bool, error) {
+					containerStatus := pod.Status.ContainerStatuses[0]
+					return containerStatus.State.Terminated != nil && containerStatus.State.Terminated.ExitCode == 0, nil
+				})
+				framework.ExpectNoError(err)
+
+				podSpec, err = client.Get(ctx, podSpec.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+				results := parseOutput(ctx, f, podSpec)
+				framework.ExpectNoError(results.HasNotRestarted(regular1))
+			})
+		})
+	})
+
+	ginkgo.When("A pod is running a regular container with restartPolicy=Always", func() {
+		regular1 := "regular-1"
+		regular2 := "regular-2"
+		bufferSeconds := int64(20)
+		exitSeconds := 60
+
+		originalPodSpec := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "container-image-update-always",
+			},
+			Spec: v1.PodSpec{
+				RestartPolicy: v1.RestartPolicyAlways,
+				Containers: []v1.Container{
+					{
+						Name:            regular1,
+						Image:           agnhostImage,
+						ImagePullPolicy: v1.PullAlways,
+						Command: ExecCommand(regular1, execCommand{
+							Delay:    exitSeconds,
+							ExitCode: 0,
+						}),
+					},
+					{
+						Name:            regular2,
+						Image:           agnhostImage,
+						ImagePullPolicy: v1.PullAlways,
+						Command: ExecCommand(regular1, execCommand{
+							Delay:    exitSeconds,
+							ExitCode: 0,
+						}),
+					},
+				},
+			},
+		}
+
+		preparePod(originalPodSpec)
+		newImage := busyboxImage
+
+		ginkgo.It("should successfully restart when the image is updated", func(ctx context.Context) {
```

Comment:
@pohly Pause does not support ExecCommand, but busybox does. We should still be fine to use agnhost in the rest of the tests in this file